sdk/core/azure-core/azure/core/serialization.py

```diff
@@ -4,20 +4,116 @@
 # Licensed under the MIT License. See License.txt in the project root for
 # license information.
 # --------------------------------------------------------------------------
+import base64
+from json import JSONEncoder
+from typing import TYPE_CHECKING
+
+from .utils._utils import _FixedOffset
+
+if TYPE_CHECKING:
+    from datetime import timedelta
 
 __all__ = ["NULL"]
 
+
 class _Null(object):
-    """To create a Falsy object
-    """
+    """To create a Falsy object"""
+
     def __bool__(self):
         return False
 
-    __nonzero__ = __bool__ # Python2 compatibility
+    __nonzero__ = __bool__  # Python2 compatibility
 
 
 NULL = _Null()
 """
 A falsy sentinel object which is supposed to be used to specify attributes
 with no data. This gets serialized to `null` on the wire.
 """
+
+
+def timedelta_as_isostr(value):
+    # type: (timedelta) -> str
+    """Converts a datetime.timedelta object into an ISO 8601 formatted string, e.g. 'P4DT12H30M05S'
+
+    Function adapted from the Tin Can Python project: https://github.com/RusticiSoftware/TinCanPython
+    """
+
+    # Split seconds to larger units
+    seconds = value.total_seconds()
+    minutes, seconds = divmod(seconds, 60)
+    hours, minutes = divmod(minutes, 60)
+    days, hours = divmod(hours, 24)
+
+    days, hours, minutes = list(map(int, (days, hours, minutes)))
+    seconds = round(seconds, 6)
+
+    # Build date
+    date = ""
+    if days:
+        date = "%sD" % days
+
+    # Build time
+    time = "T"
+
+    # Hours
+    bigger_exists = date or hours
+    if bigger_exists:
+        time += "{:02}H".format(hours)
+
+    # Minutes
+    bigger_exists = bigger_exists or minutes
+    if bigger_exists:
+        time += "{:02}M".format(minutes)
+
+    # Seconds
+    try:
+        if seconds.is_integer():
+            seconds_string = "{:02}".format(int(seconds))
+        else:
+            # 9 chars long w/ leading 0, 6 digits after decimal
+            seconds_string = "%09.6f" % seconds
+            # Remove trailing zeros
+            seconds_string = seconds_string.rstrip("0")
+    except AttributeError:  # int.is_integer() raises on Python 2.7
+        seconds_string = "{:02}".format(seconds)
```

Comment:
I wouldn't think so, since this AttributeError should only come up if `seconds` is an integer. `seconds.is_integer()` works when `seconds` is a float on 2.7. 3.6, and 3.9 when I test locally